### DISCUSSION TITLE
Any problem with "delete" and "delete_many" in array functions?

### DISCUSSION THREAD

**Discussion by claudiosa** [vlang/v#14003] [General]
Hi everyone,

I was testing some functions on arrays from: https://modules.vlang.io/#array under V interpreter (V 0.2.4 996bd41), and under these snippets, it does not works::

```
>>> import arrays
>>> mut a := ['0', '1', '2', '3', '4', '5']
>>> a.delete(4)                                     ----> remove the 5th. element
>>> print (a)
['0', '1', '2', '3', '4', '5']
>>> a.delete_many(4,1)                          ----> remove the 5th. element ... and it
>>> print (a)
['0', '1', '2', '3', '4', '5']
```

Where is my mistake? Is there any place reporting what is working and what does not work yet?
Thanks indeed

Claudio

**Comment by ntrel** [comment]
Despite delete on a map working in place, and the array delete method having a mutable receiver, it does not mutate the array in place. (There is a special exception if a flag has been set).

This is bug prone in my view, I think we should deprecate it and add other in place functions. When they were changed to return a copy the rationale was missing IMO, the one given doesn't stand scrutiny.

**Reply by ntrel** [reply]
See https://github.com/vlang/v/discussions/13878 for details.

**Reply by changrui** [reply]
It's OK when program is compiled, the problem occures under **REPL** only.

**Reply by claudiosa** [reply]
Hi @changrui and @ntrel,

Really this bug exists or it is bizarre, as well as the "pop" name function. It brings and removes the last element, not the first as expected. Look at my sample:

```
fn main() {
 
 mut a_queue := []int{}
 a_queue << [2222] //a first arrival = push
 a_queue << [21,45,67]
 a_queue << [3333] //a last arrival = a new push
 print('\n Original queue: ${a_queue}')
 mut last := a_queue.pop() // remove the LAST element NOT the first 
 mut top := a_queue[0]  // get the first
 a_queue.delete(0) // remove it

 print("\n New value of queue: ${a_queue} 
 	and last: ${last} 
	and top: ${top}\n")
}
```

The output:

```
$ v run queues_study.v 

 Original queue: [2222, 21, 45, 67, 3333]
 New value of queue: [21, 45, 67] 
 	and last: 3333 
	and top: 2222

```

Thanks indeed

**Comment by claudiosa** [comment]
@changrui and @ntrel 

My discussion above was solved (become clear) by this code on "queues":

```
fn main() {
 
 mut a_queue := []int{}
 a_queue << [2222] //a first arrival = push
 a_queue << [21,45,67]
 a_queue << [3333] //a last arrival = push
 print('\n Original queue: ${a_queue}')
 mut last := a_queue.pop() // remove the LAST element
 mut front := a_queue[0]  // get the first
 a_queue.delete(0) // remove it

 print("\n New value of queue: ${a_queue} 
 	and last: ${last} 
	and front: ${front}")

  print('\n A new departure: ${ departure(mut a_queue) }' )
  print('\n A new value of queue: ${a_queue} and lenght:  ${a_queue.len} \n' )
}

fn departure (mut queue [] int ) int {
	mut x := queue[0]
    queue.delete(0)
	//print("\n a queue in the function ${queue}")
    return x
}
```	

Any comments are very wellcome


cc
